azure-quantum/azure/quantum/target/target.py

```diff
@@ -0,0 +1,90 @@
+##
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+##
+import abc
+from typing import Any, Dict
+
+from azure.quantum.workspace import Workspace
+from azure.quantum.job.job import Job
+
+
+class Target(abc.ABC):
+    """Azure Quantum Target."""
+    def __init__(
+        self,
+        workspace: Workspace,
+        target: str,
+        input_data_format: str = "",
+        output_data_format: str = "",
+        provider_id: str = "",
+        content_type: str = "",
+        encoding: str = ""
+    ):
+        self.workspace = workspace
+        self.target = target
+        self.input_data_format = input_data_format
+        self.output_data_format = output_data_format
+        self.provider_id = provider_id
+        self.content_type = content_type
+        self.encoding = encoding
+
+    @abc.abstractstaticmethod
+    def _encode_input_data(data: Dict[Any, Any]) -> bytes:
+        """Implement abstract method to encode input data to bytes
+
+        :param data: Input data
+        :type data: Dict[Any, Any]
+        :return: Encoded input data
+        :rtype: bytes
+        """
+        pass
+
+    def submit_input_data(
```

Comment:
I don't see the value of this method. Why don't we just have `submit`?

Reply:
This method takes a `bytes` object while `submit` takes the deserialized version, e.g. JSON `dict`

Reply:
I'll make this a private method for now.